### PR TITLE
Update to 1.0 with Sylius 1.8 and Rich Editor 2.0

### PR DESCRIPTION
**Pull request by maximehuran**

### CODE REVIEW

File: .github/workflows/recipe.yaml [hunk 1]
```diff
@@ -20,7 +19,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        php: [7.2]
+        php: [7.4]
```
Comment (jacquesbh) [inline .github/workflows/recipe.yaml:22]: This should be specified in the `UPGRADE-1.0.md`, don't you think?

File: tests/Application/.env.test [hunk 2]
```diff
@@ -1,3 +1,3 @@
-APP_SECRET='s$cretf0rt3st'
+APP_SECRET='ch4mb3r0f5ecr3ts'
 
-KERNEL_CLASS='App\Kernel'
+KERNEL_CLASS='Tests\Acme\SyliusExamplePlugin\Application\Kernel'
```
Comment (jacquesbh) [inline tests/Application/.env.test:3]: Hum?

**Review by jacquesbh — CHANGES_REQUESTED** [review]
Amazing job @maximehuran ! Thank you a lot!

I pinpointed some mistakes.

File: composer.json [hunk 3]
```diff
@@ -1,29 +1,49 @@
 {
     "name": "monsieurbiz/sylius-homepage-plugin",
     "type": "sylius-plugin",
-    "keywords": ["sylius", "sylius-plugin", "homepage", "monsieurbiz"],
```
[the remark this answers is not in the document]
Comment (maximehuran) [reply composer.json:4]: To be consistent with Rich Editor and Cms page

File: LICENCE.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2020 Monsieur Biz
```
[the remark this answers is not in the document]
Comment (maximehuran) [reply LICENCE.txt:1]: I put the same as Rich Editor and CMS Page

File: composer.json [hunk 5]
```diff
@@ -1,29 +1,49 @@
...
+    "keywords": ["sylius", "sylius-plugin", "monsieurbiz"],
     "description": "Add a homepage management to your Sylius using the rich editor.",
     "license": "MIT",
     "require": {
         "php": "~7.4",
-        "ext-mbstring": "*",
```
[the remark this answers is not in the document]
Comment (maximehuran) [reply composer.json:9]: I don't think so, we are not using mb_ method in the plugin

File: src/Resources/config/sylius/resources.yaml [hunk 6]
```diff
@@ -2,11 +2,12 @@ sylius_resource:
     resources:
         monsieurbiz_homepage.homepage:
             classes:
-                model: MonsieurBiz\SyliusHomepagePlugin\Entity\Homepage\Homepage
+                model: MonsieurBiz\SyliusHomepagePlugin\Entity\Homepage
+                interface: MonsieurBiz\SyliusHomepagePlugin\Entity\HomepageInterface
                 form: MonsieurBiz\SyliusHomepagePlugin\Form\Type\HomepageType
                 repository: MonsieurBiz\SyliusHomepagePlugin\Repository\HomepageRepository
-                controller: MonsieurBiz\SyliusHomepagePlugin\Controller\HomepageController
+                controller: MonsieurBiz\SyliusHomepagePlugin\Controller\Admin\HomepageController
```
[the remark this answers is not in the document]
Comment (maximehuran) [reply src/Resources/config/sylius/resources.yaml:9]: We have to use the custom controller to use our rewrite instead of basic Resource Controller

**Comment by maximehuran** [comment]
- Added PHP MD
- Remove scrutinizer ignores
- Update PHP Stan to tun it with composer

File: composer.json [hunk 7]
```diff
@@ -1,29 +1,50 @@
...
-        "sylius/sylius": "^1.7",
-        "monsieurbiz/sylius-rich-editor-plugin": "^1.1"
+        "sylius/sylius": "^1.8",
+        "gedmo/doctrine-extensions": "^2.4.12",
+        "monsieurbiz/sylius-rich-editor-plugin": "^2.0@RC",
+        "phpstan/phpstan-webmozart-assert": "^0.12.7",
+        "phpmd/phpmd": "@stable"
```
Comment (jacquesbh) [inline composer.json:13]: This should be in require dev.

**Review by jacquesbh — CHANGES_REQUESTED** [review]
Just one change and you're good to go!

**Review by jacquesbh — APPROVED** [review]
🎉

**Comment by jacquesbh** [comment]
Amazing job @maximehuran ! 🤩

**Pull request merged**